feat: add generic BuildArgs for BuildpackArtifact

bogdannazarenko opened the pull request:
<!-- Thank you for your contribution! -->

<!-- Include if applicable: -->
Fixes: #9744 <!-- tracking issues that this PR will close -->
**Related**: N/A
**Merge before/after**: N/A

**Description**
<!-- Describe your changes here. The more detail, the easier the review! -->
Adding generic BuildArgs []string field on BuildpackArtifact. This allows users to pass any additional args to `pack build` (e.g. --network, --cache-image) without requiring future schema changes.

Example problem: GCP CloudBuild application default credentials are only available on docker network "cloudbuild," hence we need to specify that network for build pack to access that network for proper workload identity authentication for access to things such as Artifact Registries.

**User facing changes (remove if N/A)**
<!-- Describe any user facing changes this PR introduces. -->
Addition property available to set on buildpack image build configurations.
<!-- "Before" and "After" sections work great - bonus points for screenshots! -->
<!-- Be sure all docs have been updated as well! -->



<!--
Please be sure your PR includes unit tests - we don't merge code that brings down test coverage! 
Integration tests are sometimes an appropriate substitute. 
-->

bogdannazarenko commented:
@Darien-Lin @katiexzhang @menahyouyeah I generate a new schema version and added by schema changes for the buildpack build `Network` property. Could one you you please check and trigger further tests? Thank you

menahyouyeah commented on pkg/skaffold/schema/latest/config.go:
```diff
@@ -1323,6 +1323,9 @@ type BuildpackArtifact struct {
 	// Volumes support mounting host volumes into the container.
 	Volumes []*BuildpackVolume `yaml:"volumes,omitempty"`
+
+	// Network is the network to use for the build.
```
Instead of adding just the Network arg can you do something similar  to https://github.com/GoogleContainerTools/skaffold/pull/9696. Basically a BuildArgs []string or something along those lines so that in the future if people wanted to pass additional args there wouldn't need to be any additional logic or schema change.

menahyouyeah commented:
Would it be possible to split this into 2 commits - one with all the schema related changes and another with the logic changes to make it easier to review?

bogdannazarenko commented:
@Darien-Lin @katiexzhang @menahyouyeah could you please re-review when you get a chance? Thank you

Darien-Lin reviewed (changes requested):
The config struct needs to be updated to have these new fields

bogdannazarenko committed:
feat: add generic BuildArgs on BuildpackArtifact

Added generic BuildArgs []string field on BuildpackArtifact. This allows users to pass any additional args to `pack build` (e.g. --network, --cache-image) without requiring future schema changes.

Signed-off-by: Bogdan Nazarenko <bogdan.nazarenko@outlook.com>
